### ISSUE TITLE
[next-intl] Support indirect function calls with type aliases

### ISSUE THREAD

**Issue by MichaelDeBoey** [lingualdev/i18n-check#122]
In my codebase I have the following type alias:

```ts
export type NextIntlTranslateFn = ReturnType<typeof useTranslations>;
```

I then pass this type to other functions, for example:

```ts
const indirectFnCall = (t: NextIntlTranslateFn) => {
  const translation = t('key');

  // Do something with the translation
};
```

It seems that `i18n-check` currently doesn't detect translation keys used in this scenario.

Is this expected behavior?
If so, is support for this pattern planned or on the roadmap?

CC/ @busypeoples

**Comment by busypeoples** [comment]
This is expected behaviour as we only do single file static analysis due to the increase in complexity and also not knowing about the full source code (the check could be running in some sub folder).
What we can do is add an option that can be provided to define the translation function type, like `--next-intl-translation-fn` and the parser can check for that instead of `ReturnType<typeof useTranslations>`

**Comment by MichaelDeBoey** [comment]
That could work if we would support multiple ones

In my specific use-case there's both in the codebase

**Comment by busypeoples** [comment]
We will work on adding this feature for the next release.

**Comment by busypeoples** [comment]
Should be fixed in [`v0.9..4`](https://github.com/lingualdev/i18n-check/releases/tag/v0.9.4), you can try it out now.

nables to pass multiple type aliases via the `--next-intl-translation-fn-type-alias` option:

```bash
--next-intl-translation-fn-type-alias NextIntlTranslateFnAlias NextIntlTranslateFnOtherAlias
```
